docroot/modules/humsci/hs_entities/src/Entity/HsImporterType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Drupal\hs_entities\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBundleBase;
+
+/**
+ * Defines the HumSci Importer type configuration entity.
+ *
+ * @ConfigEntityType(
+ *   id = "hs_importer_type",
+ *   label = @Translation("HumSci Importer type"),
```

Comment:
this should be agnostic of importers. it will be used on content type fields for events, news, publications, etc.

Reply:
I was curious what your thoughts were gonna be on this.

Right now there are 4 different ECK Entity types (Importer collection, Publication collection, etc.,). When you create an Entity Type, you have to specify what base fields are included, whether it's revisionable or translatable, whether it has bundles or not, etc., The 4 different ECK Entity types have different base fields enabled/disabled, and I originally was going to create one Entity type for all the ECK's, but decided to just do a 1-to-1 migration from the ECK Entity type to the hard-coded Entity type because of these differences.

Reply:
To add, I'm fine with creating all these bundles under 1 Entity Type. I'm just curious what your approach would be to that.

Reply:
Also follow up just for organization:

4 ECK Entity types exist and there's only 5 bundles. 

Importer Collections
- Bundles: News RSS
- Base fields: None.

Course Collections (has Instructor bundle), 
- Bundles: Instructor
- Base fields: Label (Title), Author, Created, Changed

Event Collections
- Bundles: Conference Session, Speaker
- Base fields: Label (Title)

Publications Collections (has ) bundle).
- Bundles: Publication Author
- Base fields: Label (Title)

One issue I encountered when trying to do these as a single Entity Type was that when I added the "label" field to the News RSS bundle (that doesn't use it), I had trouble saving them because the label was required.

Reply:
All valid points. 
I don't see much concern over the Author, Created, & Changed labels. Those just get auto populated and just hang out in the DB.
For the Label property, it's only missing on the news, so we can use some mechanism to generate or just populate that label on those. For example, we use [Automatic Entity Label](https://www.drupal.org/project/auto_entitylabel) on people nodes. I wonder if that could be easily adopted to this custom entity type. If so, we could just populate it with a field value or even the current time or something. And it wouldn't change your update hook much either, just adding another property on them.

I think this way, it will reduce how much custom code we have to maintain. In the event something is deprecated during Druapl 10, we would have less to manage by going with a single entity type.

Reply:
Yea using one Entity Type sounds ideal. I do think the Automatic Entity Label solution should work for the News fields. And like you said, the other base fields are mostly in the background. I'll pivot on this and get those change in soon.

Reply:
Alright this should be done now.